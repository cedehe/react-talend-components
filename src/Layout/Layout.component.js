import React from 'react';
import classnames from 'classnames';
import AppHeaderBar from '../AppHeaderBar';
import OneColumn from './OneColumn';
import TwoColumns from './TwoColumns';
import theme from './Layout.scss';
import {
	DISPLAY_MODES,
	DISPLAY_MODE_ONE_COLUMN,
	DISPLAY_MODE_TWO_COLUMNS,
} from './constants';
/**
 * The Layout component is a container
 * that should follow the body of your App.
 * If the Layout is not after the body you have to
 * add some CSS
 * @example
body > div {
	max-height: 100vh;
	overflow: hidden;
	display: flex;
	width: 100vw;
}
 * @param {object} props react props
 * @example
 <Layout mode="TwoColumns" one={one} two={two}></Layout>
 */
function Layout({ header, mode, children, ...rest }) {
	const appCSS = classnames(
		'tc-layout',
		theme.layout
	);
	const headerCSS = classnames(
		'tc-layout-header',
		theme.header
	);
	let Component;
	switch (mode) {
	case DISPLAY_MODE_ONE_COLUMN:
		Component = OneColumn;
		break;
	case DISPLAY_MODE_TWO_COLUMNS:
		Component = TwoColumns;
		break;
	default:
		Component = OneColumn;
	}
	return (
		<div className={appCSS}>
			<div className={headerCSS}>
				<AppHeaderBar {...header} />
			</div>
			{Component ? (
				<Component {...rest}>
					{children}
				</Component>
			) : null}
		</div>
	);
}

Layout.propTypes = {
	header: React.PropTypes.shape(AppHeaderBar.propTypes),
	mode: React.PropTypes.oneOf(DISPLAY_MODES),
	children: React.PropTypes.element,
};

export default Layout;
