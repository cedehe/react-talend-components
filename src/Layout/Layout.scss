////
/// Layout
/// @group Custom widgets
////

/// Layout height of the header
/// @type Number (Unit)
$tc-layout-header-height: 49px;

body {
	margin: 0;
	max-height: 100vh;
	overflow: hidden; /* due to borders */
	width: 100vw;
}

.layout {
	display: flex;
	flex-direction: column;
	min-height: 100vh;
	width: 100vw;
}

.header {
	flex: 0 0 $tc-layout-header-height;
}
