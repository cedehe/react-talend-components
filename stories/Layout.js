import React from 'react';
import { storiesOf, action } from '@kadira/storybook';

import { Layout, SidePanel } from '../src/index';

const actions = [
	{
		label: 'Preparations',
		icon: 'fa fa-asterisk',
		onClick: action('Preparations clicked'),
	},
	{
		label: 'Datasets',
		icon: 'fa fa-file-excel-o',
		onClick: action('Datasets clicked'),
	},
	{
		label: 'Favorites',
		icon: 'fa fa-star',
		onClick: action('Favorites clicked'),
	},
];
const content = (
	<div>
		<h1>Welcome to the content for testing scroll</h1>
		<ul>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
			<li>one</li>
		</ul>
	</div>
);
const sidePanel = (<SidePanel
	actions={actions}
	onToggleDock={action('Toggle dock clicked')}
	docked={false}
/>);
const header = {
	app: 'Example App Name',
};

storiesOf('Layout', module)
	.addWithInfo('OneColumn', () => (
		<Layout
			header={header}
			mode="OneColumn"
		>
			{content}
		</Layout>
	))
	.addWithInfo('TwoColumns', () => (
		<Layout
			header={header}
			mode="TwoColumns"
			one={sidePanel}
		>
			{content}
		</Layout>
	))
	.addWithInfo('TwoColumns docked', () => {
		const dockedSidePanel = (<SidePanel
			actions={actions}
			onToggleDock={action('Toggle dock clicked')}
			docked
		/>);

		return (
			<Layout
				header={header}
				mode="TwoColumns"
				one={dockedSidePanel}
			>
				{content}
			</Layout>
		);
	});
