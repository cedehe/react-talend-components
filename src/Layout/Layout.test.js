import React from 'react';
import renderer from 'react-test-renderer';

import Layout from './Layout.component';

jest.mock('react-dom');

describe('Layout', () => {
	it('should render Layout OneColumn', () => {
		// given
		const header = { app: 'My app' };
		const one = (<h1>Column one</h1>);

		// when
		const wrapper = renderer.create(
			<Layout mode="OneColumn" one={one} header={header} />
		).toJSON();

		// then
		expect(wrapper).toMatchSnapshot();
	});

	it('should render Layout TwoColumns', () => {
		// given
		const header = { app: 'My app' };
		const one = (<h1>Column one</h1>);
		const two = (<h1>Column two</h1>);

		// when
		const wrapper = renderer.create(
			<Layout mode="TwoColumns" one={one} two={two} header={header} />
		).toJSON();

		// then
		expect(wrapper).toMatchSnapshot();
	});
});
