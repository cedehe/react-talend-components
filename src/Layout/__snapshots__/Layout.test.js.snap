exports[`Layout should render Layout OneColumn 1`] = `
<div
  className="tc-layout">
  <div
    className="tc-layout-header">
    <nav
      className="tc-app-header-bar navbar navbar-inverse navbar-fixed-top">
      <div
        className="container-fluid">
        <div
          className="navbar-header">
          <span
            className="navbar-brand">
            My app
          </span>
          <button
            className="navbar-toggle collapsed"
            onClick={[Function]}
            type="button">
            <span
              className="sr-only">
              Toggle navigation
            </span>
            <span
              className="icon-bar" />
            <span
              className="icon-bar" />
            <span
              className="icon-bar" />
          </button>
        </div>
        <div
          aria-expanded={null}
          className="navbar-collapse collapse" />
      </div>
    </nav>
  </div>
  <div
    className="tc-layout-one-column" />
</div>
`;

exports[`Layout should render Layout TwoColumns 1`] = `
<div
  className="tc-layout">
  <div
    className="tc-layout-header">
    <nav
      className="tc-app-header-bar navbar navbar-inverse navbar-fixed-top">
      <div
        className="container-fluid">
        <div
          className="navbar-header">
          <span
            className="navbar-brand">
            My app
          </span>
          <button
            className="navbar-toggle collapsed"
            onClick={[Function]}
            type="button">
            <span
              className="sr-only">
              Toggle navigation
            </span>
            <span
              className="icon-bar" />
            <span
              className="icon-bar" />
            <span
              className="icon-bar" />
          </button>
        </div>
        <div
          aria-expanded={null}
          className="navbar-collapse collapse" />
      </div>
    </nav>
  </div>
  <div
    className="tc-layout-two-columns">
    <div
      className="tc-layout-two-columns-left">
      <h1>
        Column one
      </h1>
    </div>
    <div
      className="tc-layout-two-columns-main" />
  </div>
</div>
`;
